lib/IDE/SyntaxModel.cpp

```diff
@@ -960,11 +960,13 @@ bool ModelASTWalker::walkToTypeReprPre(TypeRepr *T) {
       return false;
 
   } else if (auto IdT = dyn_cast<ComponentIdentTypeRepr>(T)) {
-    if (!passNonTokenNode({ SyntaxNodeKind::TypeId,
-                            CharSourceRange(IdT->getIdLoc(),
-                                            IdT->getIdentifier().getLength())
```

Comment:
We could use `charSourceRangeFromSourceRange(SM, IdT->getIdLoc())` too. But that invokes `Lexer::getLocForEndOfToken()` which may impact performance.